File: packages/expo-clipboard/ios/ClipboardExceptions.swift
// Copyright 2022-present 650 Industries. All rights reserved.

import ExpoModulesCore

internal class InvalidImageException: GenericException<String> {
  override var reason: String {
    "Invalid base64 image: \(param.prefix(32))\(param.count ?? 0 > 32 ? "..." : "")"
  }
}

internal class PasteFailureException: Exception {
  override var reason: String {
    "Failed to get item from clipboard"
  }
}

File: packages/expo-clipboard/android/src/main/java/expo/modules/clipboard/ClipboardExceptions.kt
package expo.modules.clipboard

import expo.modules.kotlin.exception.CodedException

internal class ClipboardUnavailableException :
  CodedException("'CLIPBOARD_SERVICE' is unavailable on this device", null)

internal class NoPermissionException(cause: SecurityException?) :
  CodedException("App has no permission to read this clipboard item", cause)

internal class PasteFailureException(cause: Throwable?, kind: String = "item") :
  CodedException("Failed to get $kind from clipboard", cause)

internal class CopyFailureException(cause: Throwable?, kind: String = "item") :
  CodedException("Failed to save $kind into clipboard", cause)

internal class InvalidImageException(image: String, cause: Throwable?) :
  CodedException(
    "Invalid base64 image: ${
    image.run { substring(0, minOf(length, 32)) + if (length > 32) "..." else ""}
    }",
    cause
  )
